mlir/lib/Interfaces/ControlFlowInterfaces.cpp

```diff
@@ -177,9 +177,10 @@ LogicalResult detail::verifyTypesAlongControlFlowEdges(Operation *op) {
 
     SmallVector<RegionBranchTerminatorOpInterface> regionReturnOps;
     for (Block &block : region)
-      if (auto terminator = dyn_cast<RegionBranchTerminatorOpInterface>(
-              block.getTerminator()))
-        regionReturnOps.push_back(terminator);
+      if (block.mightHaveTerminator())
+        if (auto terminator = dyn_cast<RegionBranchTerminatorOpInterface>(
+                block.getTerminator()))
```

Comment:
Could spell this more directly/cheaply:
```suggestion
      if (!block.empty())
        if (auto terminator = dyn_cast<RegionBranchTerminatorOpInterface>(
                block.back()))
```